### ISSUE TITLE
Crash error in Xcode8 swift3

### ISSUE THREAD

**Issue by wangbourne** [AssistoLab/DropDown#47]
it crashes if you directly use a local variable "let dropDown = DropDown()", rather than assigning dropDown as a property. However it works in Xcode7.3.

**Comment by leizard** [comment]
Seems like I have the same issue. Could you please tell more about "assigning dropDown as a property". I don't quite get that.

**Comment by leizard** [comment]
<img width="1166" alt="screen shot 2016-09-29 at 18 15 52" src="https://cloud.githubusercontent.com/assets/5291692/18979629/e89d5c12-86f6-11e6-8844-ffee280f760b.png">
This is the error that I had. Is it the same to yours ?

**Comment by wangbourne** [comment]
yes, exactly the same

**Comment by wangbourne** [comment]
by the way, if you have codes in extension via objc_setAssociatedObject, you will not be able to dropdown list ,but no crash happens

**Comment by leizard** [comment]
I finally make it work. Instead of use `let dropDown = DropDown()` in my button's IBAction scope. I create local variable `let dropDown = DropDown()` in my View Controller. then in my button IBAction function, I just init the data source, anchor view and finally show it. Now it works like a charm :D

**Comment by kevin-hirsch** [comment]
Hello @leizard and @wangbourne,

this is not new to Xcode 8 and Swift 3, every DropDown instance has no direct `superview` and is not retained by its `anchorView` (note that `anchorView` can be nil, for the DropDown to be centered on the screen). Thus, it has to be retained by something else, usually the class instance.

**Comment by Luccifer** [comment]
Bump, same problem

**Comment by AdamLenda** [comment]
Same issue

**Comment by AdamLenda** [comment]
I've been coding for 20 years and I just spent 4 hours struggling with your code. Either your documentation or your code sucks.

**Comment by ChandraNaveen** [comment]
make your dropdown instance global.It works.

**Comment by Luccifer** [comment]
you mean public static?

**Comment by ChandraNaveen** [comment]
just make it global to your class.

**Comment by jigs611989** [comment]
Make it optional
Example:
In ViewController File
`let dropDown?`

then in viewDidLoad function

```
override func viewDidLoad() {
        super.viewDidLoad()
        dropDown = DropDown()
}
```

**Comment by Lexx0** [comment]
Hello, @kevin-hirsch 

I'm having the very same problem while flipping TableViewControllers using following method. 
None of solutions listed above helps me out. 
Please advise. 

public lazy var orderedViewControllers: [UIViewController] = {
        return [self.controller(cls: FirstPageTableViewControllerStatic.self),
                self.controller(cls: SecondPageTableViewController.self)]
    }() as! [UITableViewController]

**Comment by HemantRMali** [comment]
Make it optional let dropDown: DropDown?

**Comment by kevin-hirsch** [comment]
Hello everyone!
Sorry for the late response.
I've pushed a new commit that should fix this problem (80a125b on master)
